### PR TITLE
#37205 Endpoint to check whether an upload process has been linked to…

### PR DESCRIPTION
**Pull request by Klaas-Ritense**
… a case definition

### CODE REVIEW

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessResource.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        val processDocumentDefinitions = processDocumentAssociationService.findAllProcessDocumentDefinitions(
+            CamundaProcessDefinitionKey(
+                UPLOAD_DOCUMENT_PROCESS_DEFINITION_KEY
+            )
+        )
+
+        val processCaseLinkExists = processDocumentDefinitions.any {
+            caseDefinitionKey == it.processDocumentDefinitionId().documentDefinitionId().name()
+        }
+        return ResponseEntity.ok(CheckLinkResponse(processCaseLinkExists))
```
Comment (theo-ritense) [inline zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessResource.kt:47]: What if we rename the mapping to `/case/{caseDefinitionKey}/link` and respond with either a 200 or 404?

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessAutoConfiguration.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Bean
+    @ConditionalOnMissingBean(UploadProcessResource::class)
+    fun uploadProcessResource(
+        processDocumentAssociationService: ProcessDocumentAssociationService
+    ): UploadProcessResource {
+        return UploadProcessResource(processDocumentAssociationService)
+    }
+
+    @Order(360)
```
Comment (theo-ritense) [inline zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessAutoConfiguration.kt:50]: Why is the order needed?

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessResource.kt (hunk 1 above)
Comment (Klaas-Ritense) [reply zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessResource.kt:47]: @mritense Would a 404 response work for the FE?

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessAutoConfiguration.kt (hunk 2 above)
Comment (Klaas-Ritense) [reply zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessAutoConfiguration.kt:50]: Im not sure. We have the @Order above every HttpSecurityConfigurer and the order does matter. See:
https://github.com/valtimo-platform/valtimo-backend-libraries/blob/development/core/src/main/java/com/ritense/valtimo/autoconfigure/HttpSecurityAutoConfiguration.java#L109

But I dont know how to decide which order is the correct one

File: zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessResource.kt (hunk 1 above)
Comment (Klaas-Ritense) [reply zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessResource.kt:47]: I just asked Maxim and the FE recognizes the 404 response as an error and will show toastr error popup. But, the toastr error can be intercepted. 

Still it more easy for the FE to respond with a boolean

Comment (theo-ritense) [reply zgw/zaken-api/src/main/kotlin/com/ritense/zakenapi/uploadprocess/UploadProcessResource.kt:47]: ok

**Review by theo-ritense — APPROVED** [review]

**Pull request merged**